job-server/src/spark.jobserver/WebApi.scala

```diff
@@ -61,6 +64,14 @@ object WebApi {
     Map(ResultKey -> result)
   }
 
+  def resultToByteIterator(jobReport: Map[String, Any], result: Iterator[_]): Iterator[_] = {
+    ResultKeyStartBytes.toIterator ++
+      (jobReport.map(t => Seq(AnyJsonFormat.write(t._1).toString(),
```

Comment:
Lazy is good, but in this case, your code is simply manually creating the json map bytes, then turning those into iterator anyways.... what I'm saying is that doing `AnyJsonFormat.write(jobReport)`, converting to string/bytes, then `toIterator`, is no less lazy than what you currently have (since the main lazy part is the stream output).